### PR TITLE
Customer reservations

### PR DESCRIPTION
**Pull request by juhakujala**
localhost_3000_customers_10

### CODE REVIEW

File: src/utils/mapReservationData.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export const groupReservationsByProject = (reservations: CustomerReservation[]) => {
+  if (!reservations || !reservations.length) return [];
+
+  // Use project_uuid to group each reservation under one project
+  const reservationsByProjectUUID = (): any[] => {
+    const grouped = reservations.reduce((r: any, reservation: CustomerReservation) => {
+      r[reservation.project_uuid] = r[reservation.project_uuid] || [];
+      r[reservation.project_uuid].push(reservation);
+      return r;
+    }, {});
```
Comment (tuomas777) [inline src/utils/mapReservationData.ts:12]: Hmm, wouldn't something like `const grouped = reservations.group_by(({ project_uuid }) => project_uuid);` do the same?

File: src/components/apartment/ApartmentRow.tsx [hunk 2]
```diff
@@ -222,8 +222,10 @@ const ApartmentRow = ({ apartment, ownershipType, lotteryCompleted }: IProps): J
   };
 
   const renderReservations = () => {
+    const noApplicants = () => !reservations || reservations.length === 0;
```
[the remark this answers is not in the document]
Comment (juhakujala) [reply src/components/apartment/ApartmentRow.tsx:225]: No reason, I'll change it 👍

File: src/components/customers/CustomerReservations.tsx [hunk 3]
```diff
@@ -1,93 +1,146 @@
...
+  }
+
+  // Sort reservation groups alphabetically by project name
+  const sortedReservationsByProject = [...reservationsByProject];
+  sortedReservationsByProject.sort((a, b) => a[0].project_housing_company.localeCompare(b[0].project_housing_company));
+
+  return (
+    <>
+      {sortedReservationsByProject.map((project, index) => (
+        <ReservationsByProject key={index} customer={customer} reservations={project} />
```
[the remark this answers is not in the document]
Comment (juhakujala) [reply src/components/customers/CustomerReservations.tsx:51]: That is indeed a better name. I did struggle with naming in this PR 😅

File: src/utils/mapReservationData.ts (hunk 1 above)
Comment (juhakujala) [reply src/utils/mapReservationData.ts:12]: Discussed about this:

`Array.prototype.groupBy()` is not yet natively supported by all major browsers and we probably don't want to install lodash just for this.

**Pull request merged**